synapse/storage/databases/main/devices.py

```diff
@@ -1526,6 +1526,70 @@ def _txn(txn: LoggingTransaction) -> int:
 
         return rows
 
+    async def check_too_many_devices_for_user(self, user_id: str) -> Collection[str]:
+        """Check if the user has a lot of devices, and if so return the set of
+        devices we can prune.
+
+        This does *not* return hidden devices or devices with E2E keys.
+        """
+
+        num_devices = await self.db_pool.simple_select_one_onecol(
+            table="devices",
+            keyvalues={"user_id": user_id, "hidden": False},
+            retcol="COALESCE(COUNT(*), 0)",
+            desc="count_devices",
+        )
+
+        # We let users have up to ten devices without pruning.
+        if num_devices < 10:
```

Comment:
I think this was the intention?
```suggestion
        # We let users have up to ten devices without pruning.
        if num_devices <= 10:
```